syntax = "proto3";

import "google/protobuf/empty.proto";

package FT_VDSSE;

service RPC {

// Search
rpc search (SearchRequestMessage) returns (stream SearchReply) {}

// Update
rpc update (UpdateRequestMessage) returns ( ExecuteStatus ) {}
rpc update2 (UpdateRequestMessage) returns ( ExecuteStatus ) {}
rpc batch_update (stream UpdateRequestMessage) returns ( ExecuteStatus ) {}

}

message SearchRequestMessage
{
    bytes sw = 1;
    bytes st = 2;
	uint32 c1 = 3;
    bool first = 4;
}


message SearchReply
{
	bytes ind = 1;
    bytes proof = 2;

}


message UpdateRequestMessage
{
    bytes l = 1;
    bytes e = 2;
    bytes proof = 3;
}

message ExecuteStatus
{
	bool status = 1; 
}
